src/dicarlo_lab_to_nwb/thresholding_pipeline/convert_project.py

```diff
@@ -0,0 +1,225 @@
+        "f_low": 300.0,  # Low cutoff frequency for the bandpass filter
+        "f_high": 6000.0,  # High cutoff frequency for the bandpass filter
+        "noise_threshold": 3,  # Threshold for detection in the thresholding algorithm
+    }
+
+    # Ten bins starting 200 ms before the stimulus and spanning 400 ms
+    psth_kwargs = {"bins_span_milliseconds": 500, "num_bins": 50, "milliseconds_from_event_to_first_bin": -200.0}
+    # psth_kwargs = {"bins_span_milliseconds": 400, "num_bins": 10, "milliseconds_from_event_to_first_bin": -200.0}
+
+    # This is the ground truth time column for the stimuli in the mworks csv file
+    ground_truth_time_column = "samp_on_us"
+
+    # project-level information
+    project_info = parsed_result.get("project", {})
+    project = project_info.get("name")
+    subject = project_info.get("subject")
+    nwb_output_folder = project_info.get("nwb_output_folder")
+    project_version = project_info.get("project_version")
+    pipeline_version = project_info.get("pipeline_version")
+    mworks_version = project_info.get("mworks_version")
+
+    # Merged NWB file will be saved in the nwb_output_folder
+    nwb_output_folder = Path(nwb_output_folder)
+    nwb_output_folder.mkdir(parents=True, exist_ok=True)
+
+    # experiment-level information
+    experiments = project_info.get("experiments", [])
+    for experiment in experiments:
+        experiment_date = experiment.get("date")
+        # intermediate NWB files from each day will be saved in the nwb_output_folder
+        experiment_output_folder = nwb_output_folder / f"{experiment_date}"
+        experiment_output_folder.mkdir(parents=True, exist_ok=True)
+        print(f"\nExperiment Date: {experiment_date}-------------------\n")
+
+        # session-level information
+        sessions = experiment.get("sessions", [])
+        for session in sessions:
+            stimulus_name = session.get("stimulus_name")
+            stimulus_folder = Path(session.get("stimulus_folder"))
+            mworks_folder = Path(session.get("mworks_folder"))
+            data_folder = Path(session.get("data_folder"))
+            intan_file_path = data_folder / "info.rhd"
+            assert intan_file_path.exists(), f"{intan_file_path} does not exist"
+            notes = session.get("notes")
+
+            print(f"  Session Stimulus Name: {stimulus_name}")
+            print(f"  - Stimulus Folder: {stimulus_folder}")
+            print(f"  - MWorks Folder: {mworks_folder}")
+            print(f"  - Data Folder: {data_folder}")
+            print(f"  - Notes: {notes}\n")
+
+            # session info
+            session_metadata = {
+                "project_name": project,
+                "subject": subject,
+                "recording_id": data_folder.name,
+                "pipeline_version": pipeline_version,
+            }
+
+            # Extract stimulus/behavioral events from MWorks files
+            mworks_processed_file_path = parse_mworks_file(
```

Comment:
Nice. We could add an interface that handles all of this if you want. But this also gives more flexibility so probably not necessary.

Reply:
Agreed. Let's put a pin on this for now and revisit down the road, especially if we get to expand the approach for spike sorting. For spike sorting, e.g., kilosort, we might need an interface to handle a lot more user inputs